### PR TITLE
ticlient: check nil when Backoff set vars

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?
when use `CopClient.Send`, we may pass vars with nil. `Backoff.vars` will be set to nil and then panic will occur when using it.

```go
func (c *CopClient) Send(ctx context.Context, req *kv.Request, vars *kv.Variables) kv.Response {
	bo := NewBackoffer(ctx, copBuildTaskMaxBackoff).WithVars(vars)
```

### What is changed and how it works?
check nil

### Check List 
 - No code

Code changes
 - Has exported function/method change

Related changes
 - Need to cherry-pick to the release branch

### CODE REVIEW

File: store/tikv/backoff.go [hunk 1]
```diff
@@ -196,7 +196,11 @@ func NewBackoffer(ctx context.Context, maxSleep int) *Backoffer {
 
 // WithVars sets the kv.Variables to the Backoffer and return it.
 func (b *Backoffer) WithVars(vars *kv.Variables) *Backoffer {
-	b.vars = vars
+	if vars == nil {
```
Comment (zimulala) [inline store/tikv/backoff.go]: Under what case will `vars` value be nil?

File: store/tikv/backoff.go [hunk 2]
```diff
@@ -196,7 +196,11 @@ func NewBackoffer(ctx context.Context, maxSleep int) *Backoffer {
...
+		b.vars = kv.DefaultVars
+	} else {
+		b.vars = vars
+	}
```
Comment (zimulala) [inline store/tikv/backoff.go]: I think the following code is enough：
```
if vars != nil {
	b.vars = vars
}
```

**Comment by Connor1996** [comment]
@zimulala there is no case use `vars==nil` in TiDB, but in go-ycsb we use `CopClient::Send` directly with `vars ==nil`. Maybe it is a misuse, but anyway TiDB should check nil.

**Pull request merged**